src/nft/components/collection/Card.tsx

```diff
@@ -240,16 +240,59 @@ const ImageContainer = ({ children, isDisabled = false }: { children: ReactNode;
   <StyledImageContainer isDisabled={isDisabled}>{children}</StyledImageContainer>
 )
 
-/* -------- CARD IMAGE -------- */
+const handleUniformAspectRatio = (
+  uniformAspectRatio: UniformAspectRatio,
+  e: React.SyntheticEvent<HTMLElement, Event>,
+  setUniformAspectRatio?: (uniformAspectRatio: UniformAspectRatio) => void,
+  renderedHeight?: number,
+  setRenderedHeight?: (renderedHeight: number | undefined) => void
+) => {
+  if (uniformAspectRatio !== UniformAspectRatios.square && setUniformAspectRatio) {
+    const height = e.currentTarget.clientHeight
+    const width = e.currentTarget.clientWidth
+    const ar = parseFloat((width / height).toFixed(1))
+
+    if (
+      (!renderedHeight || renderedHeight !== height) &&
+      ar < 1 &&
+      uniformAspectRatio !== UniformAspectRatios.square &&
+      setRenderedHeight
+    ) {
+      setRenderedHeight(height)
+    }
+
+    if (uniformAspectRatio === UniformAspectRatios.unset) {
+      setUniformAspectRatio(ar >= 1 ? UniformAspectRatios.square : ar)
+    } else if (uniformAspectRatio !== ar) {
+      setUniformAspectRatio(UniformAspectRatios.square)
+      setRenderedHeight && setRenderedHeight(undefined)
+    }
+  }
+}
 
-const Image = () => {
+interface ImageProps {
+  uniformAspectRatio: UniformAspectRatio
+  setUniformAspectRatio?: (uniformAspectRatio: UniformAspectRatio) => void
+  renderedHeight?: number
+  setRenderedHeight?: (renderedHeight: number | undefined) => void
+}
+
+const Image = ({ uniformAspectRatio, setUniformAspectRatio, renderedHeight, setRenderedHeight }: ImageProps) => {
   const { hovered, asset } = useCardContext()
   const [noContent, setNoContent] = useState(!asset.smallImageUrl && !asset.imageUrl)
   const [loaded, setLoaded] = useState(false)
   const isMobile = useIsMobile()
 
   if (noContent) {
-    return <NoContentContainer />
+    return (
+      <NoContentContainer
+        height={
+          uniformAspectRatio === UniformAspectRatios.square || uniformAspectRatio === UniformAspectRatios.unset
+            ? undefined
+            : renderedHeight
```

Comment:
we use this logic several times, can we make it part of the `NoContentContainer` constructor?

Reply:
Made it a helper fn